src/main/java/uk/gov/companieshouse/api/accounts/model/validation/Error.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.companieshouse.api.accounts.model.validation;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a validation error
+ */
+public class Error {
+
+    @JsonProperty("error")
+    @Field("error")
+    private String error;
+
+    @JsonProperty("error_values")
+    @Field("error_values")
+    private Map<String, String> errorValues;
+
+    @JsonProperty("location")
+    @Field("location")
+    private String location;
+
+    @JsonProperty("location_type")
+    @Field("location_type")
+    private String locationType;
+
+    @JsonProperty("type")
+    @Field("type")
+    private String type;
+
+    /**
+     * Constructor
+     *
+     * @param error
+     * @param location
+     * @param locationType
+     * @param type
+     * @throws IllegalArgumentException on null or empty arguments
+     */
+    public Error(String error, String location, String locationType, String type) {
+        if (error == null || error.isEmpty()) {
+            throw new IllegalArgumentException("Error cannot be null or empty");
+        }
+        if (location == null || location.isEmpty()) {
+            throw new IllegalArgumentException("Location cannot be null or empty");
+        }
+        if (locationType == null || locationType.isEmpty()) {
+            throw new IllegalArgumentException("Location type cannot be null or empty");
+        }
+        if (type == null || type.isEmpty()) {
+            throw new IllegalArgumentException("Type cannot be null or empty");
+        }
+
+        this.error = error;
+        this.location = location;
+        this.locationType = locationType;
+        this.type = type;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
If we are going to auto-generate `equals` and `hashCode` methods can we use the Java standard template it is a lot easier to read and will be more closely aligned to our coding standards.

Reply:
This is the same method as used in the abridged project